entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EMController.java

```diff
@@ -235,58 +245,38 @@ public EMController(EntityRecordService entityRecordService,
-	    @RequestParam(value = WebEntityConstants.QUERY_PARAM_PROFILE, defaultValue = "external") String profile,
-	    @RequestParam(value = WebEntityConstants.QUERY_PARAM_LANGUAGE, required = false) String languages,
-	    @PathVariable(value = WebEntityConstants.PATH_PARAM_TYPE) String type,
-	    @PathVariable(value = WebEntityConstants.PATH_PARAM_IDENTIFIER) String identifier,
-	    HttpServletRequest request) throws EuropeanaApiException, HttpException{
-			if (emConfig.isAuthEnabled()) {
-				verifyReadAccess(request);
-			}
-	if (acceptHeader.contains(HttpHeaders.CONTENT_TYPE_APPLICATION_RDF_XML)) {
-	    //if rdf/XML is explicitly requested
-	    return createResponse(profile, type, languages, identifier, FormatTypes.xml, HttpHeaders.CONTENT_TYPE_APPLICATION_RDF_XML);
-	}
-	if (acceptHeader.contains(MediaType.APPLICATION_XML_VALUE)) {
-	    //if XML is explicitly requested
-	    return createResponse(profile, type, languages, identifier, FormatTypes.xml, MediaType.APPLICATION_XML_VALUE);
-	} else {	
-	    //otherwise return default
-	    return createResponse(profile, type, languages, identifier, FormatTypes.jsonld, null);
-	}
-    }
+
 
     @ApiOperation(value = "Retrieve a known entity", nickname = "getEntityXml", response = java.lang.Void.class)
     @RequestMapping(value = { "/{type}/base/{identifier}.xml",
-	    "/{type}/{identifier}.xml" }, method = RequestMethod.GET, produces = {
-		    HttpHeaders.CONTENT_TYPE_APPLICATION_RDF_XML, HttpHeaders.CONTENT_TYPE_RDF_XML,
-		    MediaType.APPLICATION_XML_VALUE })
+	    "/{type}/{identifier}.xml", "/{type}/{identifier}" }, method = RequestMethod.GET, produces = {
+			MediaType.APPLICATION_XML_VALUE,
+		    HttpHeaders.CONTENT_TYPE_APPLICATION_RDF_XML, HttpHeaders.CONTENT_TYPE_RDF_XML})
     public ResponseEntity<String> getXmlEntity(
+			@RequestHeader(value = HttpHeaders.ACCEPT) String acceptHeader,
 	    @RequestParam(value = CommonApiConstants.PARAM_WSKEY, required = false) String wskey,
 	    @RequestParam(value = WebEntityConstants.QUERY_PARAM_PROFILE, defaultValue = "external") String profile,
 	    @RequestParam(value = WebEntityConstants.QUERY_PARAM_LANGUAGE, required = false) String languages,
 	    @PathVariable(value = WebEntityConstants.PATH_PARAM_TYPE) String type,
 	    @PathVariable(value = WebEntityConstants.PATH_PARAM_IDENTIFIER) String identifier,
 				HttpServletRequest request)
 				throws EuropeanaApiException, HttpException {
-			if (emConfig.isAuthEnabled()) {
+		if (emConfig.isAuthEnabled()) {
 				verifyReadAccess(request);
 			}
-	return createResponse(profile, type, languages, identifier, FormatTypes.xml, HttpHeaders.CONTENT_TYPE_APPLICATION_RDF_XML);
+			// if a valid Accept header value is provided, use it; otherwise return application/rdf+xml
+		    // TODO: handle browser requests with multiple Accept header values
+			String responseContentType = GET_XML_ACCEPT_HEADERS.contains(acceptHeader)
```

Comment:
I think that we need to return always application/rdf+xml in content type header, even when the request comes with application/xml